doc/function.md

```diff
@@ -1,5 +1,255 @@
+
+Creates instance of a `Function` object.
+
+```cpp
+template <typename Callable>
+static Function New(napi_env env, Callable cb, const char* utf8name = nullptr, void* data = nullptr);
+```
+
+- `[in] env`: The `napi_env` environment in which to construct the Function object.
+- `[in] cb`: Object that implement the operator **()** accepting a `const CallbackInfo&`
+and returns either void or Value.
+- `[in] utf8name`: Null-terminated string to be used as the name of the function.
+- `[in] data`: User-provided data context. This will be passed back into the
+function when invoked later.
+
+Returns instance of a `Function` object.
+
+### New
+
+```cpp
+template <typename Callable>
+static Function New(napi_env env, Callable cb, const std::string& utf8name, void* data = nullptr);
+```
+
+- `[in] env`: The `napi_env` environment in which to construct the Function object.
+- `[in] cb`: Object that implement the operator **()** accepting a `const CallbackInfo&`
+and returns either void or Value.
+- `[in] utf8name`: String to be used as the name of the function.
+- `[in] data`: User-provided data context. This will be passed back into the
+function when invoked later.
+
+Returns instance of a `Function` object.
+
+### Call
+
+Calls a Javascript function from a native add-on.
+
+```cpp
+Value Call(const std::initializer_list<napi_value>& args) const;
+```
+
+- `[in] args`: Initializer list of JavaScript values as `napi_value` representing
+the arguments of the function.
+
+Returns a Value representing the JavaScript object returned by the function.
+
+### Call
+
+Calls a Javascript function from a native add-on.
+
+```cpp
+Value Call(const std::vector<napi_value>& args) const;
+```
+
+- `[in] args`: List of JavaScript values as `napi_value` representing the
+arguments of the function.
+
+Returns a Value representing the JavaScript object returned by the function.
+
+### Call
+
+Calls a Javascript function from a native add-on.
+
+```cpp
+Value Call(size_t argc, const napi_value* args) const;
+```
+
+- `[in] argc`: The number of the arguments passed to the function.
+- `[in] args`: Array of JavaScript values as `napi_value` representing the
+arguments of the function.
+
+Returns a Value representing the JavaScript object returned by the function.
+
+### Call
+
+Calls a Javascript function from a native add-on.
+
+```cpp
+Value Call(napi_value recv, const std::initializer_list<napi_value>& args) const;
+```
+
+- `[in] recv`: The `this` object passed to the called function.
+- `[in] args`: Initializer list of JavaScript values as `napi_value` representing
+the arguments of the function.
+
+Returns a Value representing the JavaScript object returned by the function.
+
+### Call
+
+Calls a Javascript function from a native add-on.
+
+```cpp
+Value Call(napi_value recv, const std::vector<napi_value>& args) const;
+```
+
+- `[in] recv`: The `this` object passed to the called function.
+- `[in] args`: List of JavaScript values as `napi_value` representing the
```

Comment:
Maybe 

Vector of JavaScript values as `napi_value` representing the

Reply:
Done.